packages/core/src/stateUtils.ts

```diff
@@ -259,15 +259,49 @@ export const isStateId = (str: string) => str[0] === STATE_IDENTIFIER;
 
 export function getCandidates<TEvent extends EventObject>(
   stateNode: StateNode<any, TEvent>,
-  eventName: TEvent['type'] | NullEvent['type'] | '*'
+  eventName: TEvent['type'] | NullEvent['type'],
+  /**
+   * If `true`, will use SCXML event token matching semantics
+   */
+  matchTokens: boolean = false
 ): Array<TransitionDefinition<any, TEvent>> {
   const transient = eventName === NULL_EVENT;
   const candidates = stateNode.transitions.filter((transition) => {
-    const sameEventType = transition.eventType === eventName;
-    // null events should only match against eventless transitions
-    return transient
-      ? sameEventType
-      : sameEventType || transition.eventType === WILDCARD;
+    // First, check the trivial case: event names are exactly equal
+    if (transition.eventType === eventName) {
+      return true;
+    }
+
+    // Then, check if transition is a wildcard transition,
+    // which matches all non-transient events
+    if (transition.eventType === WILDCARD) {
+      return !transient;
+    }
+
+    if (!matchTokens && !transition.eventType.endsWith('.*')) {
```

Comment:
According to event descriptors section in the SCXML: https://www.w3.org/TR/scxml/#EventDescriptors it seems that `foo`, `foo.*` and `foo.` are all equivalent o.O which I have just learned a second ago.

Reply:
I'm not really sure if we need to support this - it's obscure and I can't find any test that would actually check this, nor I can't find this being explicitly mentioned in the CCXML spec. My vote would be against implementing support for this ;p

Reply:
Yeah, I agree with not supporting `foo.`